README.md

```diff
@@ -45,15 +45,84 @@ This demo contains below applications.
-     • Deploying application to Prod DR Environment.
-     • Publishing API’s in 3scale : (By using Operator or ansible playbooks).
+The following instructions assuming that you are using OpenShift. But the same could be applied to Minishift as well.
 
+Download the source codes from git repository by either forking, or simply cloning it. 
 
+```
+git clone https://github.com/RHsyseng/IntegrationApp-Automation.git  
+```
+Assume you have OpenShift cluster ready and running.
+
+```
+# login to the OpenShift
+oc login <OpenShift cluster url> --token=<OpenShift user login token>
+
+```
+
+Setup `rh-dev`, `rh-test` and `rh-prod` OpenShift projects as the target environment (you may skip this step if you already have the environment ready, and this script will first delete the original projects in OpenShift and create the new ones).
+    
+```
+./setup/setup.sh
+```
+
+Import the pipeline templates into your target project. For this case, it is `rh-dev`.
+
+```
+# switch to rh-dev project
+oc project rh-dev
+
+# import fisuser-service pipeline
+oc new-app -f fisuser-service/src/main/resources/pipeline-app-build.yml -p IMAGE_REGISTRY=docker-registry.default.svc:5000 -p IMAGE_NAMESPACE=rh-dev
+
+# import maingateway-service pipeline
+oc new-app -f maingateway-service/src/main/resources/pipeline-app-build.yml -p IMAGE_REGISTRY=docker-registry.default.svc:5000 -p IMAGE_NAMESPACE=rh-dev
+
+# import nodejsalert-ui pipeline
+oc new-app -f nodejsalert-ui/resources/pipeline-app-build.yml -p IMAGE_REGISTRY=docker-registry.default.svc:5000 -p IMAGE_NAMESPACE=rh-dev
+
+# import fisalert-service pipeline
+oc new-app -f fisalert-service/src/main/resources/pipeline-app-build.yml -p IMAGE_REGISTRY=docker-registry.default.svc:5000 -p IMAGE_NAMESPACE=rh-dev
+
+# import integration-master-pipeline
+oc new-app -f pipelinetemplates/pipeline-aggregated-build.yml -p IMAGE_REGISTRY=docker-registry.default.svc:5000 -p IMAGE_NAMESPACE=rh-dev
+
+```
+
+You can also customize the pipeline by changing the following parameters:
+
+```
+Parameters           Default                            Explanations
+GIT_REPO             https://github.com/RHsyseng/IntegrationApp-Automation.git
+GIT_BRANCH           master                             #git branch where you want to build
+OPENSHIFT_HOST       <leave it for now, will be used in future release>
+OPENSHIFT_TOKEN      <leave it for now, will be used in future release>
+MODULE_NAME          <module_name>                      #the Java/Node.js module name for this template to build, use default would be fine.
+DEV_PROJECT          rh-dev                
+TEST_PROJECT         rh-test
+PROD_PROJECT         rh-prod
+MYSQL_USER           dbuser                             #your DB user account
+MYSQL_PWD            password                           #DB user password
+IMAGE_REGISTRY       docker-registry.default.svc:5000   #image registry, often is the one in your OpenShift where you do the build
```

Comment:
#image registry, typically the docker registry service of your OpenShift environment

Reply:
edited